sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java

```diff
@@ -52,6 +59,19 @@ public ScreenshotEventProcessor(
       return event;
     }
 
+    // skip capturing in case of debouncing (=too many frequent capture requests)
+    // the BeforeCaptureCallback may overrules the debouncing decision
+    final boolean shouldDebounce = debouncer.checkForDebounce();
+    final @Nullable SentryAndroidOptions.BeforeCaptureCallback beforeCaptureCallback =
+        options.getBeforeScreenshotCaptureCallback();
+    if (beforeCaptureCallback != null) {
```

Comment:
just checking if that's what we want - if the `beforeCaptureCallback` is set, we will *never* debounce (or rather - we completely move this responsibility to our users). Maybe it's worth to give more details in the javadoc at least? (or maybe docs, not sure).

Reply:
Yes, exactly that's intentional - the user should have full control over this - let me update the docs then.